pilot/pkg/xds/filters/filters.go

```diff
@@ -251,38 +253,41 @@ var (
 							"workload_discovery": map[string]any{},
 						},
 					},
-					"upstream_propagation": []any{
-						map[string]any{
-							"istio_headers": map[string]any{},
-						},
-					},
 				}),
 		},
 	}
-	// TODO https://github.com/istio/istio/issues/46740
-	// false values can be omitted in protobuf, results in diff JSON values between control plane and envoy config dumps
-	// long term fix will be to add the metadata config to istio/api and use that over TypedStruct
-	SidecarOutboundMetadataFilterSkipHeaders = &hcm.HttpFilter{
-		Name: MxFilterName,
+
+	InjectIstioHeaders = &hcm.HttpFilter{
```

Comment:
Any performance impact? I'd imagine not but would be good to check

Reply:
Don't think so, but we can check and probably need both ways temporarily.